test/translate_c.zig

```diff
@@ -2704,4 +2704,20 @@ pub fn addCases(cases: *tests.TranslateCContext) void {
         \\    return if (x > y) x else y;
         \\}
     });
+
+    cases.add("Make sure casts are grouped",
+        \\typedef struct
+        \\{
+        \\    int i;
+        \\}
+        \\*_XPrivDisplay;
+        \\typedef struct _XDisplay Display;
+        \\#define DefaultScreen(dpy) 	(((_XPrivDisplay)(dpy))->default_screen)
+        \\
+    , &[_][]const u8{
+        \\pub inline fn DefaultScreen(dpy: var) @TypeOf((if (@typeInfo(@TypeOf(dpy)) == .Pointer) @ptrCast(_XPrivDisplay, @alignCast(@alignOf(_XPrivDisplay.Child), dpy)) else if (@typeInfo(@TypeOf(dpy)) == .Int) @intToPtr(_XPrivDisplay, dpy) else @as(_XPrivDisplay, dpy)).*.default_screen) {
+        \\    return (if (@typeInfo(@TypeOf(dpy)) == .Pointer) @ptrCast(_XPrivDisplay, @alignCast(@alignOf(_XPrivDisplay.Child), dpy)) else if (@typeInfo(@TypeOf(dpy)) == .Int) @intToPtr(_XPrivDisplay, dpy) else @as(_XPrivDisplay, dpy)).*.default_screen;
```

Comment:
This test will fail because I added now redundant grouping to the lhs of the arrow operator in #4660.

Reply:
Sorry about that @Vexu, I should have waited for your review here. Thanks for the fixes

Reply:
Sorry, my bad (and thank you)

Reply:
This seems to have caused breakage in master branch, so I reverted both this and @Vexu's follow-up.

Reply:
Hmm, odd. What broke? I'm able to build and run `./zig build test-run-translated-c test-translate-c` on the commit before the revert, and can build and run an X11 app.